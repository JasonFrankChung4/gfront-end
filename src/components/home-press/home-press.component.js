import React, { Component } from 'react'
import { map } from '@code.gov/cautious'

export default class HomePressComponent extends Component {
  render() {
    const bannerStyle = {
      backgroundImage: `url('${this.props.backgroundImage}')`
    }
    return (
      <section className="press" id="press-section">
        <div id="press-banner" className="banner medium" style={bannerStyle}>
          <div className="banner-content">
            <div className="indented">
              <div className="banner-title">Press</div>
              <div className="press-container">
                <div className="quote light">
                  <blockquote>{this.props.quote}</blockquote>
                  <div className="attribution">
                    <a href={this.props.attributionUrl}>
                      <img className="press-logo" src={this.props.attributionImage} alt="Wired" />
                    </a>
                  </div>
                </div>
                <div>
                  <hr />
                  <ul className="press-links">
                    {map(this.props.links, ({ image, url, alt }) => (
                      <li key={alt}>
                        <a href={url}>
                          <img className="press-logo" src={image} alt={alt} />
                        </a>
                      </li>
                    ))}
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    )
  }
}
