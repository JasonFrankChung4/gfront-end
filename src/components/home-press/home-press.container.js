import { connect } from 'react-redux'
import { getConfigValue } from 'utils/other'
import HomeFeaturedProjects from './home-press.component'

export const mapStateToProps = () => ({
  backgroundImage: getConfigValue('images.background'),
  quote: getConfigValue('content.home.press.quote') || '',
  attributionImage: getConfigValue('content.home.press.attribution.image') || '',
  attributionUrl: getConfigValue('content.home.press.attribution.url') || '',
  links: getConfigValue('content.home.press.links')
})

export default connect(mapStateToProps)(HomeFeaturedProjects)
