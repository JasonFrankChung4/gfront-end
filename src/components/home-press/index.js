import HomePressContainer from './home-press.container'

export default HomePressContainer
