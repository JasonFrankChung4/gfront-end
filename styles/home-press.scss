@import '~@code.gov/code-gov-style/_sass/_variables';

#press-section {
  .banner-content {
    transform: none;
    @media screen and (min-width:$medium-screen) {
      transform: translateY(-50%);
    }
  }

  #press-banner {
    background-attachment: "initial";
    background-position: top center;
    background-repeat: no-repeat;
    background-size: cover; // HACK: This is a hack to prevent ugliness on iOS
    @media screen and (min-width: 768px) {
      background-attachment: fixed;
    }

    .press-container {
      @include clearfix;
      font-size: 1rem;
      margin: 0 auto;
      max-width: 23em;

      @media screen and (min-width: $medium-screen) {
        font-size: 2rem;
        max-width: 250rem;
      }

      .press-links {
        margin-top: 2em;
        text-align: center;

        @media screen and (min-width: $medium-screen) {
          margin-top: 0;
        }

        li {
          display: inline-block;
          margin-bottom: 0.5em;
          margin-left: 2rem;

          &:first-of-type {
            margin-left: 0;
          }

          .press-logo {
            backface-visibility: visible;
            max-height: 3em;
            transition: all 0.2s ease;
            transform-style: preserve-3d;
            transform: translate3d(0, 0, 0);
            -webkit-perspective: 1000;

            &:hover {
              transform: translateY(-2px);
            }

            @media screen and (min-width: $small-screen) {
              max-height: 4em;
            }
          }
        }
      }
    }
  }
}
