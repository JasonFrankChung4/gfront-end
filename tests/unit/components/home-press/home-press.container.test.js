import { mapStateToProps } from 'components/home-press/home-press.container'

describe('containers - HomePress', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps()).toMatchSnapshot()
    })
  })
})
