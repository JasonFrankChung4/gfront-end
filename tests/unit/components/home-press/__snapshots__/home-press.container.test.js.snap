// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomePress mapStateToProps should return the correct properties 1`] = `
Object {
  "attributionImage": "/assets/img/logos/press/wired.svg",
  "attributionUrl": "https://www.wired.com/2016/08/open-source-won-now/",
  "backgroundImage": "/assets/img/american_flag_1920x1280_gradient.jpg",
  "links": Array [
    Object {
      "alt": "techcrunch",
      "image": "/assets/img/logos/press/techcrunch.jpg",
      "url": "https://techcrunch.com/2016/08/08/the-white-house-just-released-the-federal-source-code-policy-to-help-government-agencies-go-open-source/",
    },
    Object {
      "alt": "pc magazine",
      "image": "/assets/img/logos/press/pcmag.png",
      "url": "http://www.pcmag.com/news/346860/white-house-releases-open-source-software-policy",
    },
    Object {
      "alt": "federal computer week",
      "image": "/assets/img/logos/press/fcw.png",
      "url": "https://fcw.com/articles/2016/08/08/open-source-rockwell.aspx",
    },
  ],
  "quote": "Itâ€™s the latest in a long line of high-profile victories for the open source movement.",
}
`
